src/measure_theory/l1_space.lean

```diff
@@ -0,0 +1,181 @@
+  integrable f → integrable g → integrable (f + g) :=
+assume hfi hgi,
+  calc
+    (∫⁻ (a : α), ↑(nnnorm ((f + g) a))) ≤ ∫⁻ (a : α), ↑(nnnorm (f a)) + ↑(nnnorm (g a)) :
+      lintegral_le_lintegral _ _
+        (assume a, by { simp only [coe_add.symm, coe_le_coe], exact nnnorm_triangle _ _ })
+    ... = _ :
+      lintegral_nnnorm_add hfm hgm
+    ... < ⊤ : add_lt_top.2 ⟨hfi, hgi⟩
+
+lemma lintegral_nnnorm_neg {f : α → γ} (hf : measurable f) :
+  (∫⁻ (a : α), ↑(nnnorm ((-f) a))) = ∫⁻ (a : α), ↑(nnnorm ((f) a)) :=
+lintegral_congr_ae $ by { filter_upwards [], simp }
+
+lemma integrable_neg {f : α → γ} (hfm : measurable f) : integrable f → integrable (-f) :=
+assume hfi, calc _ = _ : lintegral_nnnorm_neg hfm
+                 ... < ⊤ : hfi
+
+section normed_space
+variables {K : Type*} [normed_field K] [normed_space K γ]
+
+lemma integrable_smul {c : K} {f : α → γ} (hfm : measurable f) : integrable f → integrable (c • f) :=
+begin
+  simp only [integrable], assume hfi,
+  calc
+    (∫⁻ (a : α), nnnorm ((c • f) a)) = (∫⁻ (a : α), (nnnorm c) * nnnorm (f a)) : by
+    { apply lintegral_congr_ae, filter_upwards [], assume a, simp [nnnorm_smul] }
+    ... < ⊤ :
+    begin
+      rw lintegral_const_mul, apply mul_lt_top,
+      { simp }, { exact hfi }, { exact measurable_coe_nnnorm hfm }
+    end
+end
+
+end normed_space
+
+namespace ae_eq_fun
+
+def integrable (f : α →ₘ γ) : Prop := f ∈ ball (0 : α →ₘ γ) ⊤
+
+@[simp] lemma integrable_mk (f : α → γ) (hf : measurable f) :
+  (integrable (mk f hf)) ↔ measure_theory.integrable f :=
+by simp [integrable, zero_def, comp_nndist, measure_theory.integrable, nndist_eq_nnnorm]
+
+lemma integrable_zero : integrable (0 : α →ₘ γ) := mem_ball_self coe_lt_top
+
+lemma integrable_add : ∀ {f g : α →ₘ γ}, integrable f → integrable g → integrable (f + g) :=
+begin
+  rintros ⟨f, hf⟩ ⟨g, hg⟩,
+  have := measure_theory.integrable_add hf hg,
+  simpa [mem_ball, zero_def]
+end
+
+lemma integrable_neg : ∀ {f : α →ₘ γ}, integrable f → integrable (-f) :=
+by { rintros ⟨f, hf⟩, have := measure_theory.integrable_neg hf, simpa }
+
+instance : is_add_subgroup (ball (0 : α →ₘ γ) ⊤) :=
+{ zero_mem := integrable_zero,
+  add_mem := λ _ _, integrable_add,
+  neg_mem := λ _, integrable_neg }
+
+section normed_space
+variables {K : Type*} [normed_field K] [second_countable_topology K] [normed_space K γ]
+
+lemma integrable_smul : ∀ {c : K} {f : α →ₘ γ}, integrable f → integrable (c • f) :=
+by { assume c, rintros ⟨f, hf⟩, have := integrable_smul hf, simpa }
+
+end normed_space
+
+end ae_eq_fun
+
+section
+variables (α γ)
+
+def l1 : Type* := subtype (@ae_eq_fun.integrable α _ γ _ _)
+
+local notation `L¹` := l1
+
+infixr ` →₁ `:25 := l1
+
+end
+
+namespace l1
+open ae_eq_fun
+
+section normed_group
+
+def mk (f : α → γ) : measurable f → integrable f → (α →₁ γ) :=
+assume hfm hfi, ⟨mk f hfm, by { rw integrable_mk, assumption }⟩
+
+@[simp] protected lemma eq {f g : α →₁ γ} : f = g ↔ f.1 = g.1 := ⟨congr rfl , subtype.eq⟩
```

Comment:
There are many lemmas that you tag as simp lemmas while they make things more complicated in my opinion. The rule for good simp lemmas is that the right hand side should be "obviously" more simple than the left hand side, which is not the case here. I would remove all the simp attributes.

Reply:
Also, I think there is an implicit(?) naming convention to call these type of lemmas `ext_iff`.